### ISSUE TITLE
Make is:exotic in LO show every build that includes one exotic

### ISSUE THREAD

**Issue by TMMania** [DestinyItemManager/DIM#6951]
Currently if the user enters `is:blue` into LO and the user doesn't have a blue item in a slot, a legendary will be utilized there instead. Example pictured below. This piece doesn't fit the user's search criteria, but LO will still complete the build with a piece that doesn't apply to the search criteria.
![image](https://user-images.githubusercontent.com/74565769/129464895-dd6feaf5-3190-40ab-addc-6626658f9074.png)

This functionality would allow the user to type in `is:exotic` into the search bar and instantly pull up their top stat tier builds utilizing any exotic. Personally, my main Grandmaster armor set was found in a very similar manner.

**Comment by bhollis** [comment]
That's not exactly how it works - each bucket is filtered, and if there are no results the filter is removed. Much later, the single exotic rule is applied across buckets, which gets rid of a lot of builds. Filtering with is:exotic says "make builds with all exotics" which can't work.

**Comment by savathunderous** [comment]
Yeah, like @bhollis explained, this issue is mixing things up a bit. There is `is:legendary` and `is:rare`, but obviously `is:exotic` doesn't work because it's selecting only exotics and cannot create a loadout with all exotics. It would be nice to have a way to select loadouts with 1 exotic though, right?? Don't show loadouts with legendaries only. Maybe have a special setting somewhere? Or since `is:exotic` doesn't make sense, maybe change its meaning in the LO? Like a special case.

**Comment by bhollis** [comment]
Can you explain what you're trying to accomplish by finding loadouts that include any exotic? Without selecting a specific exotic because of its perk, it's just like any other item in terms of stats.

**Comment by savathunderous** [comment]
Because it's an easy way to compare loadouts if you're not 100% set on one specific exotic. Just as you would compare different loadouts for one exotic, why not look for a specific stat and decide which of the available exotics suits best. Does that not make sense? It would be a hassle to select each exotic individually. And if you don't select any exotic at all, you get lots of legendary loadouts in between when you already know that you definitely want to use an exotic for its perk's sake.

**Comment by bhollis** [comment]
So to be clear you're looking for any high stat loadout with any exotic, but don't want higher stats but no exotic?

**Comment by savathunderous** [comment]
> So to be clear you're looking for any high stat loadout with any exotic, but don't want higher stats but no exotic?

Exactly.

It's seems like an easy task at first, but afaik there is currently no way to do this, right?

**Comment by bhollis** [comment]
No there isn't, and it would be quite difficult to implement as well.

**Comment by savathunderous** [comment]
> No there isn't, and it would be quite difficult to implement as well.

I guess it would be nice to have some kind of additional "postprocessing filter". Keep the current filter as it is to filter the inventory and to limit the loadouts that are being generated. And have the possibility to filter each loadout for things like:

- Is it an exotic loudout?
- Is it a tier = X loadout?
- ...

I'm sure there are more postprocessing use cases that I'm just forgetting about right now.

Is that what you meant as well when you said "quite difficult to implement" or were you thinking about something different?

**Comment by savathunderous** [comment]
> No there isn't, and it would be quite difficult to implement as well.

I guess it would be nice to have some kind of additional "postprocessing filter". Keep the current filter as it is to filter the inventory and to limit the loadouts that are being generated. And have the possibility to filter each loadout for things like:

- Is it a tier = X loadout?
- Is it an exotic loudout?
- Is it a loadout with exotic A or exotic B? (With the current preprocessing filter you can only filter for two different exotics if they're in the same slot.)
- ...

I'm sure there are more postprocessing use cases that I'm just forgetting about right now.

Is that what you meant as well when you said "quite difficult to implement" or were you thinking about something different?

**Comment by bhollis** [comment]
Yeah something like that. You'd need to plumb dow special conditions into the set generation phase, and probably add a bunch of new settings to do it, which would not only make the code more complicated but also the UI. If you're interested and think you can do it without making LO even harder for users to understand, feel free to put together a PR.

**Comment by savathunderous** [comment]
> You'd need to plumb dow special conditions into the set generation phase

I've been thinking about it a little and I'm not sure this is true. Alternatively, I might simply misunderstand what you mean with "set generation phase".

My suggestion would be to add a *post*-processing filter. So the armor sets for the LO get filtered (pre-processing) and generated exactly the same way as before, but before finalizing the results (after processing) and showing them, there is another filter that determines if the set is shown or not. For example `postprocessing:exotic` to show only loadouts with one exotic. Or `postprocessing:tier:<30`. Which both are examples for filters that are applied _after_ the loadout armor sets have been generated. It's a filter that is applied to a whole set, not to individual items.

Does it make sense what I'm saying? Or am I overlooking something?

**Comment by bhollis** [comment]
Sure, that would work.
